src/consumer/base_consumer.rs

```diff
@@ -121,6 +114,76 @@ impl<C: ConsumerContext> BaseConsumer<C> {
         unsafe { rdkafka::rd_kafka_commit_message(self.client.native_ptr(), message.ptr(), mode as i32) };
     }
 
+    /// Returns the current topic subscription.
+    pub fn subscription(&self) -> KafkaResult<TopicPartitionList> {
+        let mut tp_list = unsafe { rdkafka::rd_kafka_topic_partition_list_new(0) };
+        let error = unsafe {
+            rdkafka::rd_kafka_subscription(self.client.native_ptr(), &mut tp_list)
+        };
+
+        if error.is_error() {
+            Err(KafkaError::MetadataFetch(error))
```

Comment:
`tp_list` should be freed before returning from this branch (same in the other methods).